uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/Service/LeaveRequestTest.php

```diff
@@ -233,6 +235,47 @@ public function testDeleteSoftDeletesTheLeaveRequest() {
     $this->assertEquals(1, $leaveRequestRecord->is_deleted);
   }
 
+  public function testPublicHolidayLeaveRequestIsDeletedAndBalanceRecalculatedForOverlappingLeaveRequestDate() {
+    AbsencePeriodFabricator::fabricate([
+      'start_date' => CRM_Utils_Date::processDate('2016-01-01'),
+      'end_date' => CRM_Utils_Date::processDate('2016-12-31')
+    ]);
+
+    WorkPatternFabricator::fabricateWithA40HourWorkWeek(['is_default' => 1]);
+
+    HRJobContractFabricator::fabricate(
+      ['contact_id' => $this->leaveContact],
+      ['period_start_date' => '2016-01-01']
+    );
+
+    $publicHoliday = new PublicHoliday();
+    $publicHoliday->date = CRM_Utils_Date::processDate('2016-10-10');
+
+    $leaveRequest = LeaveRequestFabricator::fabricateWithoutValidation([
+      'from_date' => CRM_Utils_Date::processDate($publicHoliday->date),
+      'to_date' => CRM_Utils_Date::processDate($publicHoliday->date),
+      'contact_id' => $this->leaveContact,
+      'type_id' => 1,
+      'status_id' => 1
+    ], true);
+
+    $this->assertEquals(-1, LeaveBalanceChange::getTotalBalanceChangeForLeaveRequest($leaveRequest));
+
+    PublicHolidayLeaveRequestFabricator::fabricate($this->leaveContact, $publicHoliday);
+    $publicHolidayLeaveRequest = LeaveRequest::findPublicHolidayLeaveRequest($this->leaveContact, $publicHoliday);
+
+    //Balance change for Leave request will be zero since a public holiday leave request is created
+    //for same date.
+    $this->assertEquals(0, LeaveBalanceChange::getTotalBalanceChangeForLeaveRequest($leaveRequest));
+
+    $this->getLeaveRequestServiceWhenCurrentUserIsAdmin()->delete($publicHolidayLeaveRequest->id);
```

Comment:
It's probably too late to change any of this, but this test seems to be testing a lot of different services and classes. I'm guessing then that L&A uses a lot of static calls too?

Reply:
Yeah, isn't it everywhere in Civi :( ? But its limited in L&A though.